File: 0055-jump-game/0055-jump-game.c
bool canJump(int* nums, int numsSize) {
    int max_reachable = 0;

    for(int i = 0; i < numsSize; i++){
        if(max_reachable < i){
            return false;
        }

        if(max_reachable < i + nums[i]){
            max_reachable = i + nums[i];
        }

        if(max_reachable >= numsSize - 1){
            return true;
        }
    }

    return false;
}

File: 0055-jump-game/0055-jump-game.py
class Solution(object):
    def canJump(self, nums):
        max_reach = 0
        n = len(nums)
        for i in range(n):
            if i > max_reach:
                return False

            if nums[i] + i > max_reach:
                max_reach = nums[i] + i

            if max_reach >= n - 1:
                return True
        return False
        